addons/packages/contour/1.22.0/bundle/config/overlays/update-envoy-daemonset.yaml

```diff
@@ -4,9 +4,34 @@
 
 #@overlay/match by=overlay.subset({"kind": "DaemonSet", "metadata": {"name": "envoy"}})
 ---
+kind: #@ data.values.envoy.workload.type
 spec:
+  #@ if data.values.envoy.workload.type == "Deployment":
+  #@overlay/match missing_ok=True
+  replicas: #@ data.values.envoy.workload.replicas
+  #@overlay/remove
+  updateStrategy:
+  #@overlay/match missing_ok=True
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
```

Comment:
Our example Deployment uses maxSurge here instead of maxUnavailable

Reply:
noted, just changed it